File: client/src/app/inva.ts
import { enums } from './enums';

export namespace inva { 
	/* inva -  Инвентаризация */ 

 export interface   inva_info { // Описание
	inva_infoId:string; // Primary key field
	invDate:string; // Дата инвентаризации
	invReason:string; // Причина инвентаризации
	 isFinished:enums.enum_YesNo; // Инвентаризация завершена
	 isFinished_name :string; // enum to text for Инвентаризация завершена
 }

 export interface   inva_real { // Наличие
	inva_realId:string; // Primary key field
	  inva_infoId:string; // Описание
	storepartid:string; //Запчасть -> invp_data
	qty:Number; // Количество
	locationid:string; //Стеллаж -> invwh_loc
	cellid:string; //Ячейка -> invwh_cell
	theStore:string; //Склад -> invd_store
	rFID:string; // Метка RFID
	// add dereference fields 
	storepartid_name :string; // dereference for invp_data
	locationid_name :string; // dereference for invwh_loc
	cellid_name :string; // dereference for invwh_cell
	theStore_name :string; // dereference for invd_store
 }

 export interface   inva_absnt { // Недостача
	inva_absntId:string; // Primary key field
	  inva_infoId:string; // Описание
	storepartid:string; //Запчасть -> invp_data
	qty:Number; // Количество
	// add dereference fields 
	storepartid_name :string; // dereference for invp_data
 }

 export interface   inva_extra { // Излишки
	inva_extraId:string; // Primary key field
	  inva_infoId:string; // Описание
	storepartid:string; //Запчасть -> invp_data
	qty:Number; // Количество
	locationid:string; //Стеллаж -> invwh_loc
	cellid:string; //Ячейка -> invwh_cell
	rFID:string; // Метка RFID
	// add dereference fields 
	storepartid_name :string; // dereference for invp_data
	locationid_name :string; // dereference for invwh_loc
	cellid_name :string; // dereference for invwh_cell
 }
}

File: server/DataCore/inva.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace inv19.models { 
	/* inva -  Инвентраизация */ 

 public class  inva_info { // Описание
	 public System.Guid  inva_infoId{ get; set; } // Идентификатор (первичный ключ)
	 public List<inva_real>  inva_real { get; set; } // дочерний раздел: Наличие
	 public List<inva_absnt>  inva_absnt { get; set; } // дочерний раздел: Недостача
	 public List<inva_extra>  inva_extra { get; set; } // дочерний раздел: Излишки
	[Required]
	public DateTime  invDate{ get; set; } // Дата инвентаризации
	public string  invReason{ get; set; } // Причина инвентаризации
	public enum_YesNo  isFinished{ get; set; } // Инвентаризация завершена
 }

 public class  inva_real { // Наличие
	 public System.Guid  inva_realId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	 public System.Guid  inva_infoId { get; set; } // обратная ссылка на: Описание
	public System.Guid  storepartid { get; set; } //Запчасть
	
	[Required]
	public double  Qty{ get; set; } // Количество
	public System.Guid  locationid { get; set; } //Стеллаж
	
	public System.Guid  cellid { get; set; } //Ячейка
	
	public System.Guid  theStore { get; set; } //Склад
	
	[Required]
	public string  RFID{ get; set; } // Метка RFID
 }

 public class  inva_absnt { // Недостача
	 public System.Guid  inva_absntId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	 public System.Guid  inva_infoId { get; set; } // обратная ссылка на: Описание
	public System.Guid  storepartid { get; set; } //Запчасть
	
	[Required]
	public double  Qty{ get; set; } // Количество
 }

 public class  inva_extra { // Излишки
	 public System.Guid  inva_extraId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	 public System.Guid  inva_infoId { get; set; } // обратная ссылка на: Описание
	public System.Guid  storepartid { get; set; } //Запчасть
	
	[Required]
	public double  Qty{ get; set; } // Количество
	public System.Guid  locationid { get; set; } //Стеллаж
	
	public System.Guid  cellid { get; set; } //Ячейка
	
	[Required]
	public string  RFID{ get; set; } // Метка RFID
 }
}